File: Python/programmers_주식가격.py
def solution(prices):
    answer = [0 for _ in range(len(prices))]
    stack = [];
    
    for i, price in enumerate(prices):
        while len(stack) > 0 and price < prices[stack[-1]]:
            top = stack.pop();
            answer[top] = i - top;
        stack.append(i)
        
    while len(stack) > 0:
        top = stack.pop();
        answer[top] = i - top;
        
    return answer

File: JS/programmers_주식가격.js
function solution(prices) {
    const answer = new Array(prices.length).fill(0);
    const n = prices.length;
    const stack = [];
    
    for(let i=0; i<n; i++) {
        while (stack.length > 0 && prices[i] < prices[stack[stack.length - 1]]) {
            const top = stack.pop();
            answer[top] = i - top; 
        }
        stack.push(i);
    }
    
    while(stack.length > 0) {
        const top = stack.pop();
        answer[top] = n - 1 - top;
    }
        
    return answer;
}